source/common/http/http1/parser_factory.cc

```diff
@@ -0,0 +1,28 @@
+#include "common/http/http1/parser_factory.h"
+
+#include <memory>
+
+#include "common/http/http1/legacy_http_parser.h"
+#include "common/http/http1/llhttp_parser.h"
+
+namespace Envoy {
+namespace Http {
+namespace Http1 {
+
+bool ParserFactory::use_legacy_parser_ = false;
+
+ParserPtr ParserFactory::create(MessageType type, void* data) {
+  if (usesLegacyParser()) {
+    return std::make_unique<LegacyHttpParserImpl>(type, data);
+  }
+
+  return std::make_unique<LlHttpParserImpl>(type, data);
+}
+
+bool ParserFactory::usesLegacyParser() { return use_legacy_parser_; }
+
+void ParserFactory::useLegacy(bool use_legacy_parser) { use_legacy_parser_ = use_legacy_parser; }
```

Comment:
As long as we're able to link both, why not make this based on runtime?
I could imagine finding an issue where things don't work as expected, and switching all new connections over is safer than changing command line flags and doing fleet-wide restarts

Reply:
Are you suggesting _just_ runtime value and not introducing a flag? Or keeping the flag but allow runtime to override? I'm guessing the former since the latter could be confusing to someone seeing the flag enabled but not knowing the runtime value is overriding it, but just want to clarify.

Reply:
I definitely think it's worth having runtime.  Neutral about flags - last I heard from Matt most non-google companies don't use command line flags heavily so if you folks don't need it I'd lean towards sticking with one configuration method for consistency, as you say.

Reply:
Yeah I would probably just do runtime.